### DISCUSSION TITLE
GitHub Code Sync (beta)

### DISCUSSION THREAD

**Discussion by tnolet** [checkly/.github#3] [Product previews & research]
Hi all, 

We just released the public Beta for GitHub Code sync after [the private alpha](https://github.com/orgs/checkly/discussions/2) and are collecting feedback, questions and responses right here!

# What you need to know in a nutshell
- Link you Browser checks' code to a JS file in your GitHub repo.
- Sync on commit to a branch, or when a deployment event happens.
- Checkly automatically and recursively includes any local file dependencies,
- Full docs are now live [on our docs page.](https://www.checklyhq.com/docs/browser-checks/github-code-sync/)

# Examples
We are dogfooding this right now for the Checkly marketing site (a Hugo static site hosted on Vercel) and our main webapp. Check out [the public repo of our marketing site](https://github.com/checkly/checklyhq.com/tree/main/__checks__) on how we set this up.

In a nutshell
1. Create a `__check__` folder
2. Add `playwright` script in `myfile.check.js` format
3. Add any common helper and config code.
4. Link to your Browser checks in the Checkly app.

# Known limitations:
- Private repos are only available on paid plans
- This only works for Browser check scripts
- GitHub only: GitLab and/or Bitbucket not supported.
- External NPM packages are not supported outside [the packages that are part of our runtime.](https://www.checklyhq.com/docs/runtimes/specs/#npm-packages)

**Comment by keegandonley** [comment]
~~This is really nice and came at a good time. I do notice that the wording of this tooltip is a bit confusing. I have it set to sync when I push to my branch, not when a deployment completes. Is the intention that the tooltip will reflect which sync approach it's using?~~ I figured out what this means! It would be useful to me to have an indicator right here in the list view that my check is being synced from Github.

![Screen Shot 2022-08-17 at 9 43 33 AM](https://user-images.githubusercontent.com/12700819/185168524-9df4fbae-f5b3-4911-9cbd-8c4e77bdb9c1.png)

**Reply by tnolet** [reply]
@keegandonley gotcha. This icon and tooltip indeed only indicates the "trigger" behaviour, which we already had, and not the "syncing" which is new. We will have our design come up with a good option here.

**Comment by keegandonley** [comment]
Are there any plans to allow browsing of files that are `required` in a check? For example, it would be nice to be able to see the contents of these required files without having to flip back to github and find them.

<img width="795" alt="Screen Shot 2022-08-30 at 9 27 49 AM" src="https://user-images.githubusercontent.com/12700819/187463891-1b5f175b-bc0a-46a6-90ee-b50e961e8e99.png">

**Reply by kiroushi** [reply]
@keegandonley luckily we’ve just released that ability!

You will find your dependencies contents in the sidebar to the left of your editor, where the "file" icon is.

Thanks for raising your concern though, we will make sure this is more discoverable for other users.

**Comment by tnolet** [comment]
This feature is now GA! Closing the thread
